src/NHibernate/Mapping/ByCode/SimpleModelInspector.cs

```diff
@@ -31,7 +31,13 @@ public override bool IsRootEntity(System.Type entityType)
 
 			public bool IsEntity(System.Type type)
 			{
-				return RootEntities.Contains(type) || type.GetBaseTypes().Any(t => RootEntities.Contains(t)) || HasDelayedEntityRegistration(type);
+				return RootEntities.Contains(type) || type.GetBaseTypes().Any(t => RootEntities.Contains(t))
+				       || HasDelayedEntityRegistration(type);
+			}
+
+			public bool IsDeclaredEntity(System.Type type)
+			{
+				return RootEntities.Contains(type) || HasDelayedEntityRegistration(type);
```

Comment:
I doubt that HasDelayedEntityRegistration is required here. Could you please check with test that this is required?

If this is not required, then this method is equivalent to `IsRootEntity`